taxi/tests/test_views.py

```diff
@@ -0,0 +1,107 @@
+from http import client
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.forms import CarSearchForm
+from taxi.models import Manufacturer, Car
+from taxi.views import CarListView
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+CAR_LIST_URL = reverse("taxi:car-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="testname",
+            password="test1234",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="Tesla")
+        Manufacturer.objects.create(country="USA")
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(response.status_code, 200)
+        manufacturer = Manufacturer.objects.all()
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturer)
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+
+class PrivateCarTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="testname",
+            password="test1234",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_cars(self):
+        Car.objects.create(
+            model="Tesla",
+            manufacturer=Manufacturer.objects.create(
+                name="Tesla",
+                country="USA"
+            ),
+            id=1
+        )
+        Car.objects.create(
+            model="Nissan",
+            manufacturer=Manufacturer.objects.create(
+                name="Nissan",
+                country="Japan"
+            ),
+            id=2

```

Comment:
Specifying an ID during `Car` object creation is unnecessary unless you have a specific reason for doing so. The database will automatically assign an ID.